### ISSUE TITLE
[BUG] logrithmic axis allows scaling by the pan tool

### ISSUE THREAD

**Issue by wx4stg** [bokeh/bokeh#13440]
### Software versions

Python version      :  3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
IPython version     :  8.16.1
Tornado version     :  6.3.3
Bokeh version       :  3.2.2
BokehJS static path :  /home/stgardner4/micromamba/envs/acars/lib/python3.11/site-packages/bokeh/server/static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  Linux-6.5.5-200.fc38.x86_64-x86_64-with-glibc2.37

### Browser name and version

Brave Version 1.58.137 Chromium: 117.0.5938.153

### Jupyter notebook / Jupyter Lab version

jupyterlab 4.0.7

### Expected behavior

With bounds set on a figure's x/y range, attempting to click drag to pan against a bound should do nothing

### Observed behavior

When using a logarithmic y-axis, it is possible to zoom the plot by dragging against the pan limit set by the bounds, and there is no way to disable this behavior.

### Example code

```Python
from bokeh.models import Range1d, WheelZoomTool
from bokeh.plotting import figure, show
import numpy as np

x = np.linspace(1, 10, 100)
y = 10**(x/10)

fig = figure()
fig.line(x, y)

fig.x_range = Range1d(1, 10, bounds=(0, 10))
fig.y_range = Range1d(1, 10, bounds=(0, 10))

wheel = fig.select_one(WheelZoomTool)
wheel.zoom_on_axis = False
wheel.maintain_focus = False

show(fig) # dragging up or down against the vertical limits does nothing, expected behavior.

fig2 = figure(y_axis_type='log')
fig2.line(x, y)

fig2.x_range = Range1d(1, 10, bounds=(0, 10))
fig2.y_range = Range1d(1, 10, bounds=(0, 10))

wheel2 = fig2.select_one(WheelZoomTool)
wheel2.zoom_on_axis = False
wheel2.maintain_focus = False

show(fig2) # dragging vertically against the y limits zooms in on the y axis
```

### Screenshots

[Screencast from 2023-10-14 03-50-44.webm](https://github.com/bokeh/bokeh/assets/83480577/ff27dd23-972f-4ce2-b25f-ce831acfde0f)

**Comment by bryevdv** [comment]
>  attempting to click drag to pan against a bound

@wx4stg But you are not attempting to drag against the bound in this case. You have set the bound to *zero*, which is never attainable on a log axis, no matter how much panning is done. That's because the thing that are equally spaced in pixels (and thus what is actually "panned") are entire decades, and you can always make the negative exponent bigger (module fp limitations). 

I think to get any sort of lower bound on a log scale, you will need to make that bound small, but *non-zero.*

@bokeh/dev I am inclined to close this as `working as intended` but please let me know if I am missing something.

**Comment by wx4stg** [comment]
ah, yes I see that now. I first experienced this issue on a more complicated graph (a meteorological sounding which has a y-range of 1000 to 100, logrithmic) and in an attempt to dumb it down for the report and now have to include a dataset, I made the error of making one of my bounds zero which as you mentioned creates its own problems.

However, the same thing occurs when setting the lower bound to 1:

```
fig2 = figure(y_axis_type='log', title='Bug')
fig2.line(x, y)

fig2.x_range = Range1d(1, 10, bounds=(1, 10))
fig2.y_range = Range1d(1, 10, bounds=(1, 10))

wheel2 = fig2.select_one(WheelZoomTool)
wheel2.zoom_on_axis = False
wheel2.maintain_focus = False

show(fig2)
```

[Screencast from 2023-10-15 16-38-20.webm](https://github.com/bokeh/bokeh/assets/83480577/ec06b3da-31f4-49ee-b0e7-b06710a4506c)

**Comment by bryevdv** [comment]
Will require more thorough investigation for that case. [1] [This](https://github.com/bokeh/bokeh/blob/branch-3.4/bokehjs/src/lib/models/plots/range_manager.ts) is where fixed ranges, auto-ranges, log vs linear scale, constraints for aspect preservation, min and max intervals, and min/max/auto bounds, multiple axes, automatic "data following" mode, and events for every interactive tool for panning, zooming, scrolling, range tools and resetting all have to crash together.  It's an area of uncomfortably many irreducible and difficult-to-test entanglements, probably one of the gnarliest places in the whole codebase. Will take some care to make sure fixing one problem does not introduce two more...

[1] I'm not actually 100% certain this behavior was not explicitly requested, but I do agree it seems surprising offhand for now
